### ISSUE TITLE
Latest version Crash

### ISSUE THREAD

**Issue by Nekotje** [PaulCombal/SamRewritten-legacy#46]
I have downloaded the version of 19-12-2019 followed the instructions.

when i start it i get a error

> ./launch.sh 
> [PID:13814 TRD:139944960832896 TME:2.8e-05] 	PerfMon started.
> [PID:13814 TRD:139944960832896 TME:0.006912] 	Globals initialized.
> [PID:13814 TRD:139944960832896 TME:0.085055] 	Starting library parsing.
> [PID:13814 TRD:139944960832896 TME:0.577844] 	Done retrieving and filtering owned apps
> [PID:13814 TRD:139944960832896 TME:2.43412] 	Done adding apps to GUI
> 
> (samrewritten:13814): glibmm-CRITICAL **: 02:34:52.739: 
> unhandled exception (type Glib::Error) in signal handler:
> domain: gdk-pixbuf-error-quark
> code  : 3
> what  : Couldn't recognize the image file format for file '/home/user/.cache/SamRewritten/22362/banner.jpg'
> 
> [PID:13814 TRD:139944960832896 TME:6.25279] 	Starting achievement retrieval
> [S_API] SteamAPI_Init(): Loaded '/home/user/.local/share/Steam/linux64/steamclient.so' OK.
> Setting breakpad minidump AppID = 439550
> Steam_SetMinidumpSteamID:  Caching Steam ID:  [API loaded no]
> Got stats, maybe I can do cool stuff with them, gotta check.
> [PID:13814 TRD:139944960832896 TME:6.77685] 	Done retrieving achievements
> ./launch.sh: line 10: 13814 Segmentation fault      (core dumped) ${SCRIPT_PATH}/samrewritten $@

when i run ./launch it stars it sees the list of games once i open a game it crashes.

ok when i just used a different banner for the one it  breaks on i dont get any further messages untill i click a game again.

> ./launch.sh
> [PID:24283 TRD:140023187043712 TME:2.7e-05] 	PerfMon started.
> [PID:24283 TRD:140023187043712 TME:0.007215] 	Globals initialized.
> [PID:24283 TRD:140023187043712 TME:0.086904] 	Starting library parsing.
> [PID:24283 TRD:140023187043712 TME:0.564036] 	Done retrieving and filtering owned apps
> [PID:24283 TRD:140023187043712 TME:2.38064] 	Done adding apps to GUI
> [S_API] SteamAPI_Init(): Loaded '/home/user/.local/share/Steam/linux64/steamclient.so' OK.
> Setting breakpad minidump AppID = 271670
> Steam_SetMinidumpSteamID:  Caching Steam ID: [API loaded no]
> [PID:24283 TRD:140023187043712 TME:19.6135] 	Starting library parsing.
> 
> (samrewritten:24283): glibmm-ERROR **: 02:56:57.512: 
> unhandled exception (type std::exception) in signal handler:
> what: std::future_error: No associated state
> 
> ./launch.sh: line 10: 24283 Trace/breakpoint trap   (core dumped) ${SCRIPT_PATH}/samrewritten $@

**Comment by PaulCombal** [comment]
Hi, sorry to hear that.
Can you tell me what operating system you're using, what version of glibmm and gtkmm you have installed, if this is the only game affected, and if it works with the provided AppImage?
Thanks

**Comment by Nekotje** [comment]
i tried the appimage that one doesn't start at all. 
Im using Ubuntu 18.04.
With the game if it shows a different banner i don't mind i can copy a other banner in that map not sure if it overwrites it but it doesn't give an error on that.

i can create my own appimage via the package dir that one starts but has the same problem as when i do ./launch

Once i try to open a random game it crashes. 
Hope i got the correct ones.

libglib2.0-0  2.56.4-1ubuntu / libglib1.0-cil  2.12.45-0xanar
libgtkmm-3.0-1v5  3.22.2-2

**Comment by wgpierce** [comment]
Ubuntu 18.04 has older gtkmm libraries which would prevent SamRewritten from building. Ubuntu 18.04 is not officially supported, but I've created an appimage with some hacks needed to get it to work. Can you try out the Ubuntu18.04 appimage in the releases?

**Comment by Nekotje** [comment]
Same problem wgpierce.  i can get it to load but after that it just fails.  In a few months ubuntu 20.04 will be out officially will upgrade to that and in the mean time i'll wait :) thx for your help.

**Comment by Shov96** [comment]
@Nekotje I just tried out the 18.04 AppImage and it is working fine for me.

**Comment by wgpierce** [comment]
It looks like Nekotje exposes a race condition by having so many owned games, then selecting a game before the app icons are done being downloaded.


@Nekotje here's a build with some fixes in the dev branch, can you try this out?
[SamRewritten-579eaa3-x86_64.zip](https://github.com/PaulCombal/SamRewritten/files/4007556/SamRewritten-579eaa3-x86_64.zip)

**Comment by Nekotje** [comment]
> It looks like Nekotje exposes a race condition by having so many owned games, then selecting a game before the app icons are done being downloaded.
> 
> @Nekotje here's a build with some fixes in the dev branch, can you try this out?
> [SamRewritten-579eaa3-x86_64.zip](https://github.com/PaulCombal/SamRewritten/files/4007556/SamRewritten-579eaa3-x86_64.zip)

I do own 4500+ games. i tried that one also but it starts and when i try to open it just freezes the gui. 

Guess i have something changed/installed that messes with stuff. So i will wait on the ubuntu 20 and try it again than.

Ps. i did found a older version on my PC that works.

**Comment by wgpierce** [comment]
I've moved the cache folders to XDG-compliant locations, so the older version is probably picking up on the old cache. The GUI should still be responsive while the downloads are happening. Is it really non-responsive? Anything interesting in the logs?

To reconcile the cache dirs, you can move/symlink ~/.SamRewritten (old cache dir) to ~/.cache/SamRewritten (new cache dir) or similar if you set XDG variables.

**Comment by Nekotje** [comment]
> I've moved the cache folders to XDG-compliant locations, so the older version is probably picking up on the old cache. The GUI should still be responsive while the downloads are happening. Is it really non-responsive? Anything interesting in the logs?
> 
> To reconcile the cache dirs, you can move/symlink ~/.SamRewritten (old cache dir) to ~/.cache/SamRewritten (new cache dir) or similar if you set XDG variables.

if i use the commandline to start it so with ./launch i see the error and the gui just freezes  and need to be forced closed. For the appimage is the same but i dont see the error there.

**Comment by wgpierce** [comment]
> if i use the commandline to start it so with ./launch i see the error and the gui just freezes and need to be forced closed.

If you're running it from the commandline, are you using the same version as the appimage I just gave?
What's the error? Is it this one? This one should be nonfatal and is just the result of a game having a bad banner.

`what : Couldn't recognize the image file format for file '/home/user/.cache/SamRewritten/22362/banner.jpg'`

I suspect I'll have to simulate this situation to debug where the hang is occurring.

**Comment by Nekotje** [comment]
> > if i use the commandline to start it so with ./launch i see the error and the gui just freezes and need to be forced closed.
> 
> If you're running it from the commandline, are you using the same version as the appimage I just gave?
> What's the error? Is it this one? This one should be nonfatal and is just the result of a game having a bad banner.
> 
> `what : Couldn't recognize the image file format for file '/home/user/.cache/SamRewritten/22362/banner.jpg'`
> 
> I suspect I'll have to simulate this situation to debug where the hang is occurring.

 When i start ./launch
> [PID:21415 TRD:140219117872512 TME:3.4e-05] 	PerfMon started.
> [PID:21415 TRD:140219117872512 TME:0.009775] 	Globals initialized.
> [PID:21415 TRD:140219117872512 TME:0.085379] 	Starting library parsing.
> [PID:21415 TRD:140219117872512 TME:0.561164] 	Done retrieving and filtering owned apps
> [PID:21415 TRD:140219117872512 TME:2.47322] 	Done adding apps to GUI
> 
> (samrewritten:21415): glibmm-CRITICAL **: 18:18:07.568: 
> unhandled exception (type Glib::Error) in signal handler:
> domain: gdk-pixbuf-error-quark
> code  : 0
> what  : Image file '/home/user/.cache/SamRewritten/359190/banner.jpg' contains no data
> 
when i start it the gui opens and all is fine 

When i then try to open a game i get this error.
> [S_API] SteamAPI_Init(): Loaded '/home/user/.local/share/Steam/linux64/steamclient.so' OK.
> Setting breakpad minidump AppID = 299660
> Steam_SetMinidumpSteamID:  Caching Steam ID:  ID [API loaded no]
> [PID:21415 TRD:140219117872512 TME:11.59] 	Starting achievement retrieval
> Got stats, maybe I can do cool stuff with them, gotta check.
> [PID:21415 TRD:140219117872512 TME:11.7706] 	Done retrieving achievements
> ./launch.sh: line 10: 21415 Segmentation fault      (core dumped) ${SCRIPT_PATH}/samrewritten $@
> 

i got that somehow but when i try to use the make command i get this error.

> 
> sudo make
> g++ -std=c++17 -c -o obj/src/gui/AchievementBoxRow.o src/gui/AchievementBoxRow.cpp -L/home/user/Documents/SamRewritten-201912/bin -pthread -I/usr/include/gtkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gtkmm-3.0/include -I/usr/include/atkmm-1.6 -I/usr/include/gtk-3.0/unix-print -I/usr/include/gdkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gdkmm-3.0/include -I/usr/include/giomm-2.4 -I/usr/lib/x86_64-linux-gnu/giomm-2.4/include -I/usr/include/pangomm-1.4 -I/usr/lib/x86_64-linux-gnu/pangomm-1.4/include -I/usr/include/glibmm-2.4 -I/usr/lib/x86_64-linux-gnu/glibmm-2.4/include -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/cairomm-1.0 -I/usr/lib/x86_64-linux-gnu/cairomm-1.0/include -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/sigc++-2.0 -I/usr/lib/x86_64-linux-gnu/sigc++-2.0/include -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -lgtkmm-3.0 -latkmm-1.6 -lgdkmm-3.0 -lgiomm-2.4 -lpangomm-1.4 -lglibmm-2.4 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lgio-2.0 -lcairomm-1.0 -lcairo -lsigc-2.0 -lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0 -Wall -lsteam_api -lcurl -lyajl -ldl -O3
> src/gui/AchievementBoxRow.cpp: In constructor ‘AchievementBoxRow::AchievementBoxRow(const Achievement_t&)’:
> src/gui/AchievementBoxRow.cpp:33:29: error: ‘make_managed’ is not a member of ‘Gtk’
>      Gtk::Box* layout = Gtk::make_managed<Gtk::Box>(Gtk::Orientation::ORIENTATION_HORIZONTAL, 0);
>                              ^~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:33:29: note: suggested alternative: ‘manage’
>      Gtk::Box* layout = Gtk::make_managed<Gtk::Box>(Gtk::Orientation::ORIENTATION_HORIZONTAL, 0);
>                              ^~~~~~~~~~~~
>                              manage
> src/gui/AchievementBoxRow.cpp:33:50: error: expected primary-expression before ‘>’ token
>      Gtk::Box* layout = Gtk::make_managed<Gtk::Box>(Gtk::Orientation::ORIENTATION_HORIZONTAL, 0);
>                                                   ^
> src/gui/AchievementBoxRow.cpp:33:94: warning: left operand of comma operator has no effect [-Wunused-value]
>      Gtk::Box* layout = Gtk::make_managed<Gtk::Box>(Gtk::Orientation::ORIENTATION_HORIZONTAL, 0);
>                                                                                               ^
> src/gui/AchievementBoxRow.cpp:34:37: error: ‘make_managed’ is not a member of ‘Gtk’
>      Gtk::Box* title_desc_box = Gtk::make_managed<Gtk::Box>(Gtk::Orientation::ORIENTATION_VERTICAL, 0);
>                                      ^~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:34:37: note: suggested alternative: ‘manage’
>      Gtk::Box* title_desc_box = Gtk::make_managed<Gtk::Box>(Gtk::Orientation::ORIENTATION_VERTICAL, 0);
>                                      ^~~~~~~~~~~~
>                                      manage
> src/gui/AchievementBoxRow.cpp:34:58: error: expected primary-expression before ‘>’ token
>      Gtk::Box* title_desc_box = Gtk::make_managed<Gtk::Box>(Gtk::Orientation::ORIENTATION_VERTICAL, 0);
>                                                           ^
> src/gui/AchievementBoxRow.cpp:34:100: warning: left operand of comma operator has no effect [-Wunused-value]
>      Gtk::Box* title_desc_box = Gtk::make_managed<Gtk::Box>(Gtk::Orientation::ORIENTATION_VERTICAL, 0);
>                                                                                                     ^
> src/gui/AchievementBoxRow.cpp:35:36: error: ‘make_managed’ is not a member of ‘Gtk’
>      Gtk::Label* title_label = Gtk::make_managed<Gtk::Label>("");
>                                     ^~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:35:36: note: suggested alternative: ‘manage’
>      Gtk::Label* title_label = Gtk::make_managed<Gtk::Label>("");
>                                     ^~~~~~~~~~~~
>                                     manage
> src/gui/AchievementBoxRow.cpp:35:59: error: expected primary-expression before ‘>’ token
>      Gtk::Label* title_label = Gtk::make_managed<Gtk::Label>("");
>                                                            ^
> src/gui/AchievementBoxRow.cpp:36:35: error: ‘make_managed’ is not a member of ‘Gtk’
>      Gtk::Label* desc_label = Gtk::make_managed<Gtk::Label>(data.desc);
>                                    ^~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:36:35: note: suggested alternative: ‘manage’
>      Gtk::Label* desc_label = Gtk::make_managed<Gtk::Label>(data.desc);
>                                    ^~~~~~~~~~~~
>                                    manage
> src/gui/AchievementBoxRow.cpp:36:58: error: expected primary-expression before ‘>’ token
>      Gtk::Label* desc_label = Gtk::make_managed<Gtk::Label>(data.desc);
>                                                           ^
> src/gui/AchievementBoxRow.cpp:37:46: error: ‘make_managed’ is not a member of ‘Gtk’
>      Gtk::MenuButton* more_info_button = Gtk::make_managed<Gtk::MenuButton>();
>                                               ^~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:37:46: note: suggested alternative: ‘manage’
>      Gtk::MenuButton* more_info_button = Gtk::make_managed<Gtk::MenuButton>();
>                                               ^~~~~~~~~~~~
>                                               manage
> src/gui/AchievementBoxRow.cpp:37:74: error: expected primary-expression before ‘>’ token
>      Gtk::MenuButton* more_info_button = Gtk::make_managed<Gtk::MenuButton>();
>                                                                           ^
> src/gui/AchievementBoxRow.cpp:37:76: error: expected primary-expression before ‘)’ token
>      Gtk::MenuButton* more_info_button = Gtk::make_managed<Gtk::MenuButton>();
>                                                                             ^
> src/gui/AchievementBoxRow.cpp:38:40: error: ‘make_managed’ is not a member of ‘Gtk’
>      Gtk::Image* more_info_image = Gtk::make_managed<Gtk::Image>("gtk-about", Gtk::BuiltinIconSize::ICON_SIZE_BUTTON);
>                                         ^~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:38:40: note: suggested alternative: ‘manage’
>      Gtk::Image* more_info_image = Gtk::make_managed<Gtk::Image>("gtk-about", Gtk::BuiltinIconSize::ICON_SIZE_BUTTON);
>                                         ^~~~~~~~~~~~
>                                         manage
> src/gui/AchievementBoxRow.cpp:38:63: error: expected primary-expression before ‘>’ token
>      Gtk::Image* more_info_image = Gtk::make_managed<Gtk::Image>("gtk-about", Gtk::BuiltinIconSize::ICON_SIZE_BUTTON);
>                                                                ^
> src/gui/AchievementBoxRow.cpp:38:100: warning: left operand of comma operator has no effect [-Wunused-value]
>      Gtk::Image* more_info_image = Gtk::make_managed<Gtk::Image>("gtk-about", Gtk::BuiltinIconSize::ICON_SIZE_BUTTON);
>                                                                                                     ^~~~~~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:39:43: error: ‘make_managed’ is not a member of ‘Gtk’
>      Gtk::PopoverMenu* popover_menu = Gtk::make_managed<Gtk::PopoverMenu>();
>                                            ^~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:39:43: note: suggested alternative: ‘manage’
>      Gtk::PopoverMenu* popover_menu = Gtk::make_managed<Gtk::PopoverMenu>();
>                                            ^~~~~~~~~~~~
>                                            manage
> src/gui/AchievementBoxRow.cpp:39:72: error: expected primary-expression before ‘>’ token
>      Gtk::PopoverMenu* popover_menu = Gtk::make_managed<Gtk::PopoverMenu>();
>                                                                         ^
> src/gui/AchievementBoxRow.cpp:39:74: error: expected primary-expression before ‘)’ token
>      Gtk::PopoverMenu* popover_menu = Gtk::make_managed<Gtk::PopoverMenu>();
>                                                                           ^
> src/gui/AchievementBoxRow.cpp:40:34: error: ‘make_managed’ is not a member of ‘Gtk’
>      Gtk::Box* popover_box = Gtk::make_managed<Gtk::Box>(Gtk::Orientation::ORIENTATION_VERTICAL, 3);
>                                   ^~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:40:34: note: suggested alternative: ‘manage’
>      Gtk::Box* popover_box = Gtk::make_managed<Gtk::Box>(Gtk::Orientation::ORIENTATION_VERTICAL, 3);
>                                   ^~~~~~~~~~~~
>                                   manage
> src/gui/AchievementBoxRow.cpp:40:55: error: expected primary-expression before ‘>’ token
>      Gtk::Box* popover_box = Gtk::make_managed<Gtk::Box>(Gtk::Orientation::ORIENTATION_VERTICAL, 3);
>                                                        ^
> src/gui/AchievementBoxRow.cpp:40:97: warning: left operand of comma operator has no effect [-Wunused-value]
>      Gtk::Box* popover_box = Gtk::make_managed<Gtk::Box>(Gtk::Orientation::ORIENTATION_VERTICAL, 3);
>                                                                                                  ^
> src/gui/AchievementBoxRow.cpp:41:41: error: ‘make_managed’ is not a member of ‘Gtk’
>      Gtk::Label* more_info_label  = Gtk::make_managed<Gtk::Label>();
>                                          ^~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:41:41: note: suggested alternative: ‘manage’
>      Gtk::Label* more_info_label  = Gtk::make_managed<Gtk::Label>();
>                                          ^~~~~~~~~~~~
>                                          manage
> src/gui/AchievementBoxRow.cpp:41:64: error: expected primary-expression before ‘>’ token
>      Gtk::Label* more_info_label  = Gtk::make_managed<Gtk::Label>();
>                                                                 ^
> src/gui/AchievementBoxRow.cpp:41:66: error: expected primary-expression before ‘)’ token
>      Gtk::Label* more_info_label  = Gtk::make_managed<Gtk::Label>();
>                                                                   ^
> src/gui/AchievementBoxRow.cpp:42:49: error: ‘make_managed’ is not a member of ‘Gtk’
>      Gtk::Label* percentage_players_label = Gtk::make_managed<Gtk::Label>(ach_player_percent_text);
>                                                  ^~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:42:49: note: suggested alternative: ‘manage’
>      Gtk::Label* percentage_players_label = Gtk::make_managed<Gtk::Label>(ach_player_percent_text);
>                                                  ^~~~~~~~~~~~
>                                                  manage
> src/gui/AchievementBoxRow.cpp:42:72: error: expected primary-expression before ‘>’ token
>      Gtk::Label* percentage_players_label = Gtk::make_managed<Gtk::Label>(ach_player_percent_text);
>                                                                         ^
> src/gui/AchievementBoxRow.cpp:43:36: error: ‘make_managed’ is not a member of ‘Gtk’
>      Gtk::Separator* sep_one = Gtk::make_managed<Gtk::Separator>();
>                                     ^~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:43:36: note: suggested alternative: ‘manage’
>      Gtk::Separator* sep_one = Gtk::make_managed<Gtk::Separator>();
>                                     ^~~~~~~~~~~~
>                                     manage
> src/gui/AchievementBoxRow.cpp:43:63: error: expected primary-expression before ‘>’ token
>      Gtk::Separator* sep_one = Gtk::make_managed<Gtk::Separator>();
>                                                                ^
> src/gui/AchievementBoxRow.cpp:43:65: error: expected primary-expression before ‘)’ token
>      Gtk::Separator* sep_one = Gtk::make_managed<Gtk::Separator>();
>                                                                  ^
> src/gui/AchievementBoxRow.cpp:44:36: error: ‘make_managed’ is not a member of ‘Gtk’
>      Gtk::Separator* sep_two = Gtk::make_managed<Gtk::Separator>();
>                                     ^~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:44:36: note: suggested alternative: ‘manage’
>      Gtk::Separator* sep_two = Gtk::make_managed<Gtk::Separator>();
>                                     ^~~~~~~~~~~~
>                                     manage
> src/gui/AchievementBoxRow.cpp:44:63: error: expected primary-expression before ‘>’ token
>      Gtk::Separator* sep_two = Gtk::make_managed<Gtk::Separator>();
>                                                                ^
> src/gui/AchievementBoxRow.cpp:44:65: error: expected primary-expression before ‘)’ token
>      Gtk::Separator* sep_two = Gtk::make_managed<Gtk::Separator>();
>                                                                  ^
> src/gui/AchievementBoxRow.cpp:45:47: error: ‘make_managed’ is not a member of ‘Gtk’
>      Gtk::ProgressBar* ach_progress_bar = Gtk::make_managed<Gtk::ProgressBar>(); // TODO
>                                                ^~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:45:47: note: suggested alternative: ‘manage’
>      Gtk::ProgressBar* ach_progress_bar = Gtk::make_managed<Gtk::ProgressBar>(); // TODO
>                                                ^~~~~~~~~~~~
>                                                manage
> src/gui/AchievementBoxRow.cpp:45:76: error: expected primary-expression before ‘>’ token
>      Gtk::ProgressBar* ach_progress_bar = Gtk::make_managed<Gtk::ProgressBar>(); // TODO
>                                                                             ^
> src/gui/AchievementBoxRow.cpp:45:78: error: expected primary-expression before ‘)’ token
>      Gtk::ProgressBar* ach_progress_bar = Gtk::make_managed<Gtk::ProgressBar>(); // TODO
>                                                                               ^
> src/gui/AchievementBoxRow.cpp:46:47: error: ‘make_managed’ is not a member of ‘Gtk’
>      Gtk::ProgressBar* ach_ach_rate_bar = Gtk::make_managed<Gtk::ProgressBar>();
>                                                ^~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:46:47: note: suggested alternative: ‘manage’
>      Gtk::ProgressBar* ach_ach_rate_bar = Gtk::make_managed<Gtk::ProgressBar>();
>                                                ^~~~~~~~~~~~
>                                                manage
> src/gui/AchievementBoxRow.cpp:46:76: error: expected primary-expression before ‘>’ token
>      Gtk::ProgressBar* ach_ach_rate_bar = Gtk::make_managed<Gtk::ProgressBar>();
>                                                                             ^
> src/gui/AchievementBoxRow.cpp:46:78: error: expected primary-expression before ‘)’ token
>      Gtk::ProgressBar* ach_ach_rate_bar = Gtk::make_managed<Gtk::ProgressBar>();
>                                                                               ^
> src/gui/AchievementBoxRow.cpp:47:39: error: ‘make_managed’ is not a member of ‘Gtk’
>      Gtk::Label* progress_label = Gtk::make_managed<Gtk::Label>("Completed X of Y");
>                                        ^~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:47:39: note: suggested alternative: ‘manage’
>      Gtk::Label* progress_label = Gtk::make_managed<Gtk::Label>("Completed X of Y");
>                                        ^~~~~~~~~~~~
>                                        manage
> src/gui/AchievementBoxRow.cpp:47:62: error: expected primary-expression before ‘>’ token
>      Gtk::Label* progress_label = Gtk::make_managed<Gtk::Label>("Completed X of Y");
>                                                               ^
> src/gui/AchievementBoxRow.cpp:73:34: error: ‘make_managed’ is not a member of ‘Gtk’
>          Gtk::Label* label = Gtk::make_managed<Gtk::Label>("💎   ");
>                                   ^~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:73:34: note: suggested alternative: ‘manage’
>          Gtk::Label* label = Gtk::make_managed<Gtk::Label>("💎   ");
>                                   ^~~~~~~~~~~~
>                                   manage
> src/gui/AchievementBoxRow.cpp:73:57: error: expected primary-expression before ‘>’ token
>          Gtk::Label* label = Gtk::make_managed<Gtk::Label>("💎   ");
>                                                          ^
> src/gui/AchievementBoxRow.cpp:80:34: error: ‘make_managed’ is not a member of ‘Gtk’
>          Gtk::Label* label = Gtk::make_managed<Gtk::Label>("📈   ");
>                                   ^~~~~~~~~~~~
> src/gui/AchievementBoxRow.cpp:80:34: note: suggested alternative: ‘manage’
>          Gtk::Label* label = Gtk::make_managed<Gtk::Label>("📈   ");
>                                   ^~~~~~~~~~~~
>                                   manage
> src/gui/AchievementBoxRow.cpp:80:57: error: expected primary-expression before ‘>’ token
>          Gtk::Label* label = Gtk::make_managed<Gtk::Label>("📈   ");
>                                                          ^
> Makefile:28: recipe for target 'obj/src/gui/AchievementBoxRow.o' failed
> make: *** [obj/src/gui/AchievementBoxRow.o] Error 1

i saw this post

> https://stackoverflow.com/questions/53334845/why-do-i-get-the-compile-error-make-managed-is-not-a-member-of-gtk/53922602#53922602

so i checked the object.h file and  the part wasnt in my file

> template<class T, class... T_Args>
> auto make_managed(T_Args&&... args) -> T* // Note: Edited to add return type here!
> {
>   return manage(new T(std::forward<T_Args>(args)...));
> }
this part. When i add this part to the object.h file and than do a Make command i get a other error

> g++ -std=c++17 -c -o obj/src/gui/AchievementBoxRow.o src/gui/AchievementBoxRow.cpp -L/home/user/Documents/SamRewritten-201912/bin -pthread -I/usr/include/gtkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gtkmm-3.0/include -I/usr/include/atkmm-1.6 -I/usr/include/gtk-3.0/unix-print -I/usr/include/gdkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gdkmm-3.0/include -I/usr/include/giomm-2.4 -I/usr/lib/x86_64-linux-gnu/giomm-2.4/include -I/usr/include/pangomm-1.4 -I/usr/lib/x86_64-linux-gnu/pangomm-1.4/include -I/usr/include/glibmm-2.4 -I/usr/lib/x86_64-linux-gnu/glibmm-2.4/include -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/cairomm-1.0 -I/usr/lib/x86_64-linux-gnu/cairomm-1.0/include -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/sigc++-2.0 -I/usr/lib/x86_64-linux-gnu/sigc++-2.0/include -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -lgtkmm-3.0 -latkmm-1.6 -lgdkmm-3.0 -lgiomm-2.4 -lpangomm-1.4 -lglibmm-2.4 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lgio-2.0 -lcairomm-1.0 -lcairo -lsigc-2.0 -lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0 -Wall -lsteam_api -lcurl -lyajl -ldl -O3
> In file included from /usr/include/gtkmm-3.0/gtkmm/widget.h:39:0,
>                  from /usr/include/gtkmm-3.0/gtkmm/container.h:28,
>                  from /usr/include/gtkmm-3.0/gtkmm/bin.h:27,
>                  from /usr/include/gtkmm-3.0/gtkmm/listboxrow.h:25,
>                  from src/gui/ListBoxRowWithIcon.h:4,
>                  from src/gui/AchievementBoxRow.h:5,
>                  from src/gui/AchievementBoxRow.cpp:1:
> /usr/include/gtkmm-3.0/gtkmm/object.h: In instantiation of ‘T* Gtk::make_managed(T_Args&& ...) [with T = Gtk::Image; T_Args = {const char (&)[10], Gtk::BuiltinIconSize}]’:
> src/gui/AchievementBoxRow.cpp:38:116:   required from here
> /usr/include/gtkmm-3.0/gtkmm/object.h:50:17: error: no matching function for call to ‘Gtk::Image::Image(const char [10], Gtk::BuiltinIconSize)’
>    return manage(new T(std::forward<T_Args>(args)...));
>                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> In file included from src/gui/ListBoxRowWithIcon.h:5:0,
>                  from src/gui/AchievementBoxRow.h:5,
>                  from src/gui/AchievementBoxRow.cpp:1:
> /usr/include/gtkmm-3.0/gtkmm/image.h:251:12: note: candidate: Gtk::Image::Image(const Glib::RefPtr<Gdk::PixbufAnimation>&)
>    explicit Image(const Glib::RefPtr<Gdk::PixbufAnimation>& animation);
>             ^~~~~
> /usr/include/gtkmm-3.0/gtkmm/image.h:251:12: note:   candidate expects 1 argument, 2 provided
> /usr/include/gtkmm-3.0/gtkmm/image.h:248:3: note: candidate: Gtk::Image::Image(const Glib::RefPtr<Gtk::IconSet>&, Gtk::IconSize)
>    Image(const Glib::RefPtr<IconSet>& icon_set, IconSize size);
>    ^~~~~
> /usr/include/gtkmm-3.0/gtkmm/image.h:248:3: note:   no known conversion for argument 1 from ‘const char [10]’ to ‘const Glib::RefPtr<Gtk::IconSet>&’
> /usr/include/gtkmm-3.0/gtkmm/image.h:242:3: note: candidate: Gtk::Image::Image(const Gtk::StockID&, Gtk::IconSize)
>    Image(const Gtk::StockID& stock_id, IconSize size);
>    ^~~~~
> /usr/include/gtkmm-3.0/gtkmm/image.h:242:3: note:   no known conversion for argument 1 from ‘const char [10]’ to ‘const Gtk::StockID&’
> /usr/include/gtkmm-3.0/gtkmm/image.h:225:14: note: candidate: Gtk::Image::Image(const Glib::RefPtr<Gdk::Pixbuf>&)
>      explicit Image(const Glib::RefPtr<Gdk::Pixbuf>& pixbuf);
>               ^~~~~
> /usr/include/gtkmm-3.0/gtkmm/image.h:225:14: note:   candidate expects 1 argument, 2 provided
> /usr/include/gtkmm-3.0/gtkmm/image.h:218:14: note: candidate: Gtk::Image::Image(const string&)
>      explicit Image(const std::string& file);
>               ^~~~~
> /usr/include/gtkmm-3.0/gtkmm/image.h:218:14: note:   candidate expects 1 argument, 2 provided
> In file included from src/gui/ListBoxRowWithIcon.h:5:0,
>                  from src/gui/AchievementBoxRow.h:5,
>                  from src/gui/AchievementBoxRow.cpp:1:
> /usr/include/gtkmm-3.0/gtkmm/image.h:206:3: note: candidate: Gtk::Image::Image()
>    Image();
>    ^~~~~
> /usr/include/gtkmm-3.0/gtkmm/image.h:206:3: note:   candidate expects 0 arguments, 2 provided
> /usr/include/gtkmm-3.0/gtkmm/image.h:169:12: note: candidate: Gtk::Image::Image(GtkImage*)
>    explicit Image(GtkImage* castitem);
>             ^~~~~
> /usr/include/gtkmm-3.0/gtkmm/image.h:169:12: note:   candidate expects 1 argument, 2 provided
> /usr/include/gtkmm-3.0/gtkmm/image.h:168:12: note: candidate: Gtk::Image::Image(const Glib::ConstructParams&)
>    explicit Image(const Glib::ConstructParams& construct_params);
>             ^~~~~
> /usr/include/gtkmm-3.0/gtkmm/image.h:168:12: note:   candidate expects 1 argument, 2 provided
> /usr/include/gtkmm-3.0/gtkmm/image.h:152:3: note: candidate: Gtk::Image::Image(Gtk::Image&&)
>    Image(Image&& src) noexcept;
>    ^~~~~
> /usr/include/gtkmm-3.0/gtkmm/image.h:152:3: note:   candidate expects 1 argument, 2 provided
> Makefile:28: recipe for target 'obj/src/gui/AchievementBoxRow.o' failed
> make: *** [obj/src/gui/AchievementBoxRow.o] Error 1

**Comment by PaulCombal** [comment]
You can't build SamRewritten because Ubuntu 18 is too old and is not supported. Yet, how can you do ./launch.sh if you can't build 🤔? I'm not sure if I can help you with the debugging process since you are not on a supported platform.

**Comment by Nekotje** [comment]
> You can't build SamRewritten because Ubuntu 18 is too old and is not supported. Yet, how can you do ./launch.sh if you can't build thinking? I'm not sure if I can help you with the debugging process since you are not on a supported platform.

i was looking at what error it game.
so i removed 1 line and than do the make command it it gives this output.

> sudo make
> [sudo] password for user: 
> g++ -std=c++17 -c -o obj/src/gui/AchievementBoxRow.o src/gui/AchievementBoxRow.cpp -L/home/user/Documents/SamRewritten-201912/bin -pthread -I/usr/include/gtkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gtkmm-3.0/include -I/usr/include/atkmm-1.6 -I/usr/include/gtk-3.0/unix-print -I/usr/include/gdkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gdkmm-3.0/include -I/usr/include/giomm-2.4 -I/usr/lib/x86_64-linux-gnu/giomm-2.4/include -I/usr/include/pangomm-1.4 -I/usr/lib/x86_64-linux-gnu/pangomm-1.4/include -I/usr/include/glibmm-2.4 -I/usr/lib/x86_64-linux-gnu/glibmm-2.4/include -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/cairomm-1.0 -I/usr/lib/x86_64-linux-gnu/cairomm-1.0/include -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/sigc++-2.0 -I/usr/lib/x86_64-linux-gnu/sigc++-2.0/include -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -lgtkmm-3.0 -latkmm-1.6 -lgdkmm-3.0 -lgiomm-2.4 -lpangomm-1.4 -lglibmm-2.4 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lgio-2.0 -lcairomm-1.0 -lcairo -lsigc-2.0 -lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0 -Wall -lsteam_api -lcurl -lyajl -ldl -O3
> g++ -std=c++17 -c -o obj/src/schema_parser/UserGameStatsSchemaParser.o src/schema_parser/UserGameStatsSchemaParser.cpp -L/home/user/Documents/SamRewritten-201912/bin -pthread -I/usr/include/gtkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gtkmm-3.0/include -I/usr/include/atkmm-1.6 -I/usr/include/gtk-3.0/unix-print -I/usr/include/gdkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gdkmm-3.0/include -I/usr/include/giomm-2.4 -I/usr/lib/x86_64-linux-gnu/giomm-2.4/include -I/usr/include/pangomm-1.4 -I/usr/lib/x86_64-linux-gnu/pangomm-1.4/include -I/usr/include/glibmm-2.4 -I/usr/lib/x86_64-linux-gnu/glibmm-2.4/include -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/cairomm-1.0 -I/usr/lib/x86_64-linux-gnu/cairomm-1.0/include -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/sigc++-2.0 -I/usr/lib/x86_64-linux-gnu/sigc++-2.0/include -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -lgtkmm-3.0 -latkmm-1.6 -lgdkmm-3.0 -lgiomm-2.4 -lpangomm-1.4 -lglibmm-2.4 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lgio-2.0 -lcairomm-1.0 -lcairo -lsigc-2.0 -lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0 -Wall -lsteam_api -lcurl -lyajl -ldl -O3
> g++ -std=c++17 -c -o obj/src/schema_parser/KeyValue.o src/schema_parser/KeyValue.cpp -L/home/user/Documents/SamRewritten-201912/bin -pthread -I/usr/include/gtkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gtkmm-3.0/include -I/usr/include/atkmm-1.6 -I/usr/include/gtk-3.0/unix-print -I/usr/include/gdkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gdkmm-3.0/include -I/usr/include/giomm-2.4 -I/usr/lib/x86_64-linux-gnu/giomm-2.4/include -I/usr/include/pangomm-1.4 -I/usr/lib/x86_64-linux-gnu/pangomm-1.4/include -I/usr/include/glibmm-2.4 -I/usr/lib/x86_64-linux-gnu/glibmm-2.4/include -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/cairomm-1.0 -I/usr/lib/x86_64-linux-gnu/cairomm-1.0/include -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/sigc++-2.0 -I/usr/lib/x86_64-linux-gnu/sigc++-2.0/include -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -lgtkmm-3.0 -latkmm-1.6 -lgdkmm-3.0 -lgiomm-2.4 -lpangomm-1.4 -lglibmm-2.4 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lgio-2.0 -lcairomm-1.0 -lcairo -lsigc-2.0 -lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0 -Wall -lsteam_api -lcurl -lyajl -ldl -O3
> g++ -std=c++17 -o /home/user/Documents/SamRewritten-201912/bin/samrewritten obj/src/main.o obj/src/cli/cli_funcs.o obj/src/json/ProcessedGameServerRequest.o obj/src/json/yajlHelpers.o obj/src/sockets/MyClientSocket.o obj/src/sockets/MyGameSocket.o obj/src/sockets/MySocket.o obj/src/sockets/MyServerSocket.o obj/src/controller/MySteam.o obj/src/controller/SteamAppDAO.o obj/src/controller/GameServerManager.o obj/src/controller/MyGameServer.o obj/src/common/Downloader.o obj/src/common/PerfMon.o obj/src/common/functions.o obj/src/gui/AppBoxRow.o obj/src/gui/InputAppidBoxRow.o obj/src/gui/AsyncGuiLoader.o obj/src/gui/MainPickerWindowFactory.o obj/src/gui/MainPickerWindow.o obj/src/gui/ListBoxRowWithIcon.o obj/src/gui/AchievementBoxRow.o obj/src/schema_parser/UserGameStatsSchemaParser.o obj/src/schema_parser/KeyValue.o -L/home/user/Documents/SamRewritten-201912/bin -pthread -I/usr/include/gtkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gtkmm-3.0/include -I/usr/include/atkmm-1.6 -I/usr/include/gtk-3.0/unix-print -I/usr/include/gdkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gdkmm-3.0/include -I/usr/include/giomm-2.4 -I/usr/lib/x86_64-linux-gnu/giomm-2.4/include -I/usr/include/pangomm-1.4 -I/usr/lib/x86_64-linux-gnu/pangomm-1.4/include -I/usr/include/glibmm-2.4 -I/usr/lib/x86_64-linux-gnu/glibmm-2.4/include -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/cairomm-1.0 -I/usr/lib/x86_64-linux-gnu/cairomm-1.0/include -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/sigc++-2.0 -I/usr/lib/x86_64-linux-gnu/sigc++-2.0/include -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -lgtkmm-3.0 -latkmm-1.6 -lgdkmm-3.0 -lgiomm-2.4 -lpangomm-1.4 -lglibmm-2.4 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lgio-2.0 -lcairomm-1.0 -lcairo -lsigc-2.0 -lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0 -Wall -lsteam_api -lcurl -lyajl -ldl -O3
> -e ==== Use 'make dev' to keep debug symbols
> -e ==== Use 'make clean' to remove object files
> -e ==== Nothing left to do.

when i add the code 

> template<class T, class... T_Args>
> auto make_managed(T_Args&&... args) -> T* // Note: Edited to add return type here!
> {
> return manage(new T(std::forward<T_Args>(args)...));
> }

i get the error but when i remove the line

> return manage(new T(std::forward<T_Args>(args)...));

i get the output above. and i can run ./launch.sh

for the rest it doesnt matter that much i just wait a few months and upgrade my workstation to ubuntu 20 so than all should work fine.

**Comment by PaulCombal** [comment]
Alright then, let me know if you still encounter this kind of issues with Ubuntu 20! (By the way you don't need to use "sudo" to make ;) )

**Comment by wgpierce** [comment]
@Nekotje here's a debug build, it should give more info on where it's crashing, can you try this out and report the logs from it?
[SamRewritten-579eaa3-x86_64_debug.zip](https://github.com/PaulCombal/SamRewritten/files/4010958/SamRewritten-579eaa3-x86_64_debug.zip)

If you want to build it on 18.04, you need the hack you mentioned in the library plus a change to a gtk library function: 
[18.04.txt](https://github.com/PaulCombal/SamRewritten/files/4010968/18.04.txt)
Note this patch isn't officially supported

**Comment by Nekotje** [comment]
> @Nekotje here's a debug build, it should give more info on where it's crashing, can you try this out and report the logs from it?
> [SamRewritten-579eaa3-x86_64_debug.zip](https://github.com/PaulCombal/SamRewritten/files/4010958/SamRewritten-579eaa3-x86_64_debug.zip)
> 
> If you want to build it on 18.04, you need the hack you mentioned in the library plus a change to a gtk library function:
> [18.04.txt](https://github.com/PaulCombal/SamRewritten/files/4010968/18.04.txt)
> Note this patch isn't officially supported

i did download your debug version but that one seems to work. with opening games. i can open a game and see all achievements from that game.

**Comment by wgpierce** [comment]
Can you try clearing your cache and opening it a few times?

**Comment by Nekotje** [comment]
> Can you try clearing your cache and opening it a few times?

Hi i cleaned the cache from ~/home/.cache/SamRewritten a few times

and started the appimage and loaded a game  go back and load another one. it seems to work all fine  all pictures as far as i can see are working etc.

**Comment by wgpierce** [comment]
Hmm. Can you try applying the patch I mentioned and try compiling with -02 in the Makefile and seeing if running that gives you an error?

@PaulCombal Maybe -03 is producing dangerous optimizations... Also turning on -Wextra doesn't give any more warnings that will affect anything...

**Comment by Nekotje** [comment]
> Hmm. Can you try applying the patch I mentioned and try compiling with -02 in the Makefile and seeing if running that gives you an error?
> 
> @PaulCombal Maybe -03 is producing dangerous optimizations... Also turning on -Wextra doesn't give any more warnings that will affect anything...

i did 
> git apply --stat 18.04.patch
> src/gui/AchievementBoxRow.cpp |    7 ++++---
> 1 file changed, 4 insertions(+), 3 deletions(-)
> git apply --check 18.04.patch
> git am 18.04.patch

once i try the make this is what i get. ( this is the first time that i applied a patch to a file)
```
make
g++ -std=c++17 -c -o obj/src/gui/AchievementBoxRow.o src/gui/AchievementBoxRow.cpp -L/home/user/Documents/SamRewritten-201912/bin -pthread -I/usr/include/gtkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gtkmm-3.0/include -I/usr/include/atkmm-1.6 -I/usr/include/gtk-3.0/unix-print -I/usr/include/gdkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gdkmm-3.0/include -I/usr/include/giomm-2.4 -I/usr/lib/x86_64-linux-gnu/giomm-2.4/include -I/usr/include/pangomm-1.4 -I/usr/lib/x86_64-linux-gnu/pangomm-1.4/include -I/usr/include/glibmm-2.4 -I/usr/lib/x86_64-linux-gnu/glibmm-2.4/include -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/cairomm-1.0 -I/usr/lib/x86_64-linux-gnu/cairomm-1.0/include -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/sigc++-2.0 -I/usr/lib/x86_64-linux-gnu/sigc++-2.0/include -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -lgtkmm-3.0 -latkmm-1.6 -lgdkmm-3.0 -lgiomm-2.4 -lpangomm-1.4 -lglibmm-2.4 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lgio-2.0 -lcairomm-1.0 -lcairo -lsigc-2.0 -lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0 -Wall -lsteam_api -lcurl -lyajl -ldl -O2
Assembler messages:
Fatal error: can't create obj/src/gui/AchievementBoxRow.o: Permission denied
Makefile:28: recipe for target 'obj/src/gui/AchievementBoxRow.o' failed
make: *** [obj/src/gui/AchievementBoxRow.o] Error 1

sudo make
[sudo] password for user: 
g++ -std=c++17 -c -o obj/src/gui/AchievementBoxRow.o src/gui/AchievementBoxRow.cpp -L/home/user/Documents/SamRewritten-201912/bin -pthread -I/usr/include/gtkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gtkmm-3.0/include -I/usr/include/atkmm-1.6 -I/usr/include/gtk-3.0/unix-print -I/usr/include/gdkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gdkmm-3.0/include -I/usr/include/giomm-2.4 -I/usr/lib/x86_64-linux-gnu/giomm-2.4/include -I/usr/include/pangomm-1.4 -I/usr/lib/x86_64-linux-gnu/pangomm-1.4/include -I/usr/include/glibmm-2.4 -I/usr/lib/x86_64-linux-gnu/glibmm-2.4/include -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/cairomm-1.0 -I/usr/lib/x86_64-linux-gnu/cairomm-1.0/include -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/sigc++-2.0 -I/usr/lib/x86_64-linux-gnu/sigc++-2.0/include -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -lgtkmm-3.0 -latkmm-1.6 -lgdkmm-3.0 -lgiomm-2.4 -lpangomm-1.4 -lglibmm-2.4 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lgio-2.0 -lcairomm-1.0 -lcairo -lsigc-2.0 -lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0 -Wall -lsteam_api -lcurl -lyajl -ldl -O2
g++ -std=c++17 -o /home/user/Documents/SamRewritten-201912/bin/samrewritten obj/src/main.o obj/src/cli/cli_funcs.o obj/src/json/ProcessedGameServerRequest.o obj/src/json/yajlHelpers.o obj/src/sockets/MyClientSocket.o obj/src/sockets/MyGameSocket.o obj/src/sockets/MySocket.o obj/src/sockets/MyServerSocket.o obj/src/controller/MySteam.o obj/src/controller/SteamAppDAO.o obj/src/controller/GameServerManager.o obj/src/controller/MyGameServer.o obj/src/common/Downloader.o obj/src/common/PerfMon.o obj/src/common/functions.o obj/src/gui/AppBoxRow.o obj/src/gui/InputAppidBoxRow.o obj/src/gui/AsyncGuiLoader.o obj/src/gui/MainPickerWindowFactory.o obj/src/gui/MainPickerWindow.o obj/src/gui/ListBoxRowWithIcon.o obj/src/gui/AchievementBoxRow.o obj/src/schema_parser/UserGameStatsSchemaParser.o obj/src/schema_parser/KeyValue.o -L/home/user/Documents/SamRewritten-201912/bin -pthread -I/usr/include/gtkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gtkmm-3.0/include -I/usr/include/atkmm-1.6 -I/usr/include/gtk-3.0/unix-print -I/usr/include/gdkmm-3.0 -I/usr/lib/x86_64-linux-gnu/gdkmm-3.0/include -I/usr/include/giomm-2.4 -I/usr/lib/x86_64-linux-gnu/giomm-2.4/include -I/usr/include/pangomm-1.4 -I/usr/lib/x86_64-linux-gnu/pangomm-1.4/include -I/usr/include/glibmm-2.4 -I/usr/lib/x86_64-linux-gnu/glibmm-2.4/include -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/cairomm-1.0 -I/usr/lib/x86_64-linux-gnu/cairomm-1.0/include -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/sigc++-2.0 -I/usr/lib/x86_64-linux-gnu/sigc++-2.0/include -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -lgtkmm-3.0 -latkmm-1.6 -lgdkmm-3.0 -lgiomm-2.4 -lpangomm-1.4 -lglibmm-2.4 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lgio-2.0 -lcairomm-1.0 -lcairo -lsigc-2.0 -lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0 -Wall -lsteam_api -lcurl -lyajl -ldl -O2
-e ==== Use 'make dev' to keep debug symbols
-e ==== Use 'make clean' to remove object files
-e ==== Nothing left to do.
```

bin/./launch.sh
```
[PID:24161 TRD:140125332456832 TME:3.4e-05] 	PerfMon started.
[PID:24161 TRD:140125332456832 TME:0.007382] 	Globals initialized.
[PID:24161 TRD:140125332456832 TME:0.086657] 	Starting library parsing.
[PID:24161 TRD:140125332456832 TME:0.609903] 	Done retrieving and filtering owned apps
[PID:24161 TRD:140125332456832 TME:2.5308] 	Done adding apps to GUI

(samrewritten:24161): glibmm-CRITICAL **: 21:13:53.522: 
unhandled exception (type Glib::Error) in signal handler:
domain: gdk-pixbuf-error-quark
code  : 3
what  : Couldn't recognize the image file format for file '/home/user/.cache/SamRewritten/22362/banner.jpg'

[S_API] SteamAPI_Init(): Loaded '/home/user/.local/share/Steam/linux64/steamclient.so' OK.
Setting breakpad minidump AppID = 227580
Steam_SetMinidumpSteamID:  Caching Steam ID:  [API loaded no]
[PID:24161 TRD:140125332456832 TME:6.16563] 	Starting achievement retrieval
Got stats, maybe I can do cool stuff with them, gotta check.
[PID:24161 TRD:140125332456832 TME:6.36618] 	Done retrieving achievements
[PID:24161 TRD:140125332456832 TME:6.37913] 	Done adding achievements to GUI
[PID:24161 TRD:140125332456832 TME:6.39204] 	Done parsing schema to find achievement icon download names
[PID:24161 TRD:140125332456832 TME:6.51356] 	Achievements retrieved with icons.
Shutting down server safely.
[PID:24161 TRD:140125332456832 TME:7.00781] 	Starting achievement retrieval
[S_API] SteamAPI_Init(): Loaded '/home/user/.local/share/Steam/linux64/steamclient.so' OK.
Setting breakpad minidump AppID = 205690
Steam_SetMinidumpSteamID:  Caching Steam ID:   [API loaded no]
Got stats, maybe I can do cool stuff with them, gotta check.
[PID:24161 TRD:140125332456832 TME:7.2334] 	Done retrieving achievements
[PID:24161 TRD:140125332456832 TME:7.2424] 	Done adding achievements to GUI
[PID:24161 TRD:140125332456832 TME:7.25349] 	Done parsing schema to find achievement icon download names
[PID:24161 TRD:140125332456832 TME:7.28925] 	Achievements retrieved with icons.
Shutting down server safely.
[PID:24161 TRD:140125332456832 TME:8.0025] 	Starting achievement retrieval
[S_API] SteamAPI_Init(): Loaded '/home/user/.local/share/Steam/linux64/steamclient.so' OK.
Setting breakpad minidump AppID = 381780
Steam_SetMinidumpSteamID:  Caching Steam ID:  [API loaded no]
Got stats, maybe I can do cool stuff with them, gotta check.
[PID:24161 TRD:140125332456832 TME:8.19727] 	Done retrieving achievements
[PID:24161 TRD:140125332456832 TME:8.22101] 	Done adding achievements to GUI
[PID:24161 TRD:140125332456832 TME:8.24542] 	Done parsing schema to find achievement icon download names
[PID:24161 TRD:140125332456832 TME:8.28873] 	Achievements retrieved with icons.
Shutting down server safely.
```
This is all working it seems. Like said i didn't try to unlock any achievements but the program is working till now.

**Comment by PaulCombal** [comment]
Glad it works for you! I'm going to close the issue for now but feel free to open another one for another issue. 
About -O3, I didn't have any problems with it yet, so unless there is a proof this is an issue I don't think this requires attention.

**Comment by wgpierce** [comment]
If you're still on Ubuntu 18.04, the latest master includes commits 082ad7f1182e9a8e3ae9d90d82267b6f33f9b658 and 7358596158221af89b19f89f25d4ee8b4e8ecfb1 to fix these problems and enable building.
